econml/sklearn_extensions/model_selection.py

```diff
@@ -386,19 +389,31 @@ def _fit_with_groups(model, X, y, *, sub_model=None, groups, **kwargs):
 
 class FixedModelSelector(SingleModelSelector):
     """
-    Model selection class that always selects the given model
+    Model selection class that always selects the given sklearn-compatible model
     """
 
     def __init__(self, model):
         self.model = clone(model, safe=False)
 
-    def train(self, is_selecting, *args, groups=None, **kwargs):
-        # whether selecting or not, need to train the model on the data
-        _fit_with_groups(self.model, *args, groups=groups, **kwargs)
-        if is_selecting and hasattr(self.model, 'score'):
-            # TODO: we need to alter this to use out-of-sample score here, which
-            #       will require cross-validation, but should respect grouping, stratifying, etc.
-            self._score = self.model.score(*args, **kwargs)
+    def train(self, is_selecting, folds: Optional[List], X, y, groups=None, **kwargs):
+        if is_selecting:
+            # since needs_fit is False, is_selecting will only be true if
+            # the score needs to be compared to another model's
+            # so we don't need to fit the model itself, just get the out-of-sample score
+            assert hasattr(self.model, 'score'), (f"Can't select between a fixed {type(self.model)} model and others "
+                                                  "because it doesn't have a score method")
+            scores = []
+            for train, test in folds:
+                # use _fit_with_groups instead of just fit to handle nested grouping
```

Comment:
I'm a little confused here. Why do we need to do nested grouping for a FixedModelSelector?

I'm having trouble grasping why we seem to be doing two levels of splitting here. (one level within this train function, and another level within _fit_with_groups).

Couldn't we get away with only one level of splitting?

Reply:
In practice `_fit_with_groups` almost always just calls plain fit on all of the data without doing any further splitting; the exception is if there are groups and the model that we've wrapped has a `cv` attribute, in which case the model is going to be doing its own cross-validation when we call fit no matter what, so we just want to make sure that we update the `cv` attribute to respect the groups.

In the typical case, if a user passes a known sklearn model that supports cross validation, we won't wrap it in a FixedModelSelector anyway, but this gives users an escape hatch if they want to be doing cross-validation within each fold for some reason.